### PR TITLE
Correct usage of ARCH. Issue # 2530.

### PR DESCRIPTION
**Pull request by vkostas**
_Motivation:_ Issue [2530 - https://github.com/SynoCommunity/spksrc/issues/2530](https://github.com/SynoCommunity/spksrc/issues/2530)
_Linked issues:_ N/A

### Checklist
- [] Build rule `all-supported` completed successfully
- [] Package upgrade completed successfully
- [] New installation of package completed successfully

### CODE REVIEW

File: cross/alsa-lib/Makefile [hunk 1]
```diff
@@ -13,7 +13,7 @@ LICENSE  =
 
 GNU_CONFIGURE = 1
 CONFIGURE_ARGS = --disable-python
-ifeq ($(findstring $(ARCH),88f5281 ppc853x ppc854x),$(ARCH))
+ifeq ($(findstring $(ARCH),$(PPC_ARCHES)),$(ARCH))
```
Comment (Dr-Bean) [inline cross/alsa-lib/Makefile]: Missing `include ../../mk/spksrc.cross-cc.mk` before this line.
Can we agree to put the include after GNU_CONFIGURE (or LICENSE, if GNU_CONFIGURE isn't set), and before CONFIGURE_ARGS?
That approach keeps all the CONFIGURE_ARGS more or less in the same place.

File: cross/bison/Makefile [hunk 2]
```diff
@@ -12,7 +12,10 @@ COMMENT  = General-purpose parser generator
 LICENSE  = GNU GPL
 
 GNU_CONFIGURE = 1
-ifeq ($(findstring $(ARCH),88f5281 powerpc ppc824x ppc853x ppc854x),$(ARCH))
+
+include ../../mk/spksrc.cross-cc.mk
```
Comment (Dr-Bean) [inline cross/bison/Makefile]: The include on L24 should be removed.

File: cross/bitlbee/Makefile [hunk 3]
```diff
@@ -11,6 +11,8 @@ HOMEPAGE = http://www.bitlbee.org/
 COMMENT  = BitlBee brings IM (instant messaging) to IRC clients. It\'s a great solution for people who have an IRC client running all the time and don\'t want to run an additional MSN/AIM/whatever client
 LICENSE  =
 
+include ../../mk/spksrc.common.mk
```
Comment (Dr-Bean) [inline cross/bitlbee/Makefile]: I thought we agreed that we didn't want to include common.mk, but that we'd use `include ../../mk/spksrc.cross-cc.mk` instead?

File: cross/ffmpeg-2.2.1/digests [hunk 4]
```diff
@@ -0,0 +1,3 @@
+ffmpeg-2.2.1.tar.gz SHA1 cf455dd900cd7749f2c51c644054a9b657b8b589
```
Comment (Dr-Bean) [inline cross/ffmpeg-2.2.1/digests]: While this may be necessary, it doesn't belong in this PR. Can you move it to a separate one?

File: cross/alsa-lib/Makefile (hunk 1 above)
Comment (Dr-Bean) [inline cross/alsa-lib/Makefile]: And line breaks in between. So, to be clear, the generic approach would look like this
```
LICENSE...

GNU_CONFIGURE ...if applicable

include ../...

CONFIGURE_ARGS = ...if applicable

ifeq($(findstring $(ARCH)...
  <thing>
endif

.PHONY etc. ...if applicable

Comment (vkostas) [inline cross/alsa-lib/Makefile]: I was thinking of the same.
Agree!!!

File: cross/bitlbee/Makefile (hunk 3 above)
Comment (vkostas) [inline cross/bitlbee/Makefile]: My fault. :-)

File: cross/ffmpeg-2.2.1/digests (hunk 4 above)
Comment (vkostas) [inline cross/ffmpeg-2.2.1/digests]: Ok

File: cross/bitlbee/Makefile (hunk 3 above)
Comment (vkostas) [inline cross/bitlbee/Makefile]: hmmm. I think using **cross-cc.mk** before our variables at least for **TARGET_PATCH, TARGET_INSTALL, etc**.
That's why I applied added **common.mk** before the variable declarations used in **cross-cc.mk**.
I have to redo all changes again:
>LICENSE...
>
>GNU_CONFIGURE ...if applicable
>
>include ../../mk/spksrc.common.mk
>
>CONFIGURE_ARGS = ...if applicable
>
>ifeq($(findstring $(ARCH)...
>  <thing>
>endif
>
>include ../../mk/spksrc.cross-cc.mk
>
>.PHONY etc. ...if applicable
>

**Comment by vkostas** [comment]
@Dr-Bean Just in case you have missed my reply .....
Maybe the solution is as simple as to include **common.mk** at the very beginning of cross Makefile (sometimes ARCH is needed to define dependencies).
>include ../../mk/spksrc.common.mk
PKG_NAME = 
PKG_VERS = 
PKG_EXT = 
...
DEPENDS
...
...
include ../../mk/spksrc.cross-cc.mk

.PHONY
>

**Comment by Dr-Bean** [comment]
Can't, because common.mk contains optional values (e.g. the `pip ?=` entry), you wouldn't be able to define those values correctly.

Besides, let's keep this simple. common.mk is included via cross-cc.mk. For the packages where we currently need access to values of `Available Arches` (that set of arches), we already know it works when we put the include above it. There should be a general order that we can establish to put the include for those particular packages, yes? Let's keep the other packages as is for now, because it's more work than it's worth imo :)
